src/manage_nsfs/manage_nsfs_validations.js

```diff
@@ -65,6 +66,34 @@ async function validate_input_types(type, action, argv) {
     }
 }
 
+/**
+ * validate_no_leading_space ensures that flags do not contain leading spaces
+ *  - if `argv_` contains more than two elements (type and action), it indicates an invalid flag value.
+ *
+ *
+ * ### CLI Parsing Behavior:
+ * When passing arguments via CLI, if a flag's value starts with a space or consists only of whitespace,
+ * the shell may treat it differently:
+ *  - Leading spaces - might be preserved when quoted (`--new_buckets_path=" abc/"`).
+ *  - Unquoted values with spaces - may cause incorrect parsing (`--new_buckets_path= abc/` is interpreted as an empty value).
+ *  - The parsed argument structure typically follows:
+ *      ```json
+ *      { _: [ "command", "action" ], flag1: "value1", flag2: "value2", ... }
+ *      ```
+ *  - The `_` array should contain only the command and action. If additional elements appear,
+ *    it suggests an issue with flag values (e.g., an incorrectly formatted or empty value).
+ *
+ * @param {string[]} argv_
+ * @throws {ManageCLIError}
+ */
+function validate_no_leading_space(argv_) {
+    // checking if 'argv_' contains more then 2 values (i.e, type and action)
+    if (argv_.length > 2) {
+        const details = 'flag value is invalid - leading spaces or an empty value detected, ensure flag values are properly formatted.';
+        throw_cli_error(ManageCLIError.InvalidArgument, details);
```

Comment:
If you'll have a more strict check you will be able to add in the details the exact invalid value.
From you example:
on `sudo node src/cmd/manage_nsfs account add --name aayush3 --uid=9 --gid=9 --allow_bucket_creation=true --new_buckets_path= tmp/`

it is:
`_: [ 'account', 'add', 'tmp/' ]`

And you can explicitly mention the "tmp/" as the invalid value that was passed.

Reply:
make sense, will update it. Thanks